polaris-core/src/main/java/org/apache/polaris/core/storage/aws/StsClientSupplier.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.polaris.core.storage.aws;
+
+import jakarta.annotation.Nullable;
+import java.net.URI;
+import java.util.Optional;
+import org.apache.polaris.immutables.PolarisImmutable;
+import org.immutables.value.Value;
+import software.amazon.awssdk.services.sts.StsClient;
+
+public interface StsClientSupplier {
```

Comment:
Perhaps this is rather a `Factory`? At first, I thought `StsClientSupplier` would be a `Supplier<StsClient>`. Even if it's not literally this type it's also not logically this type because it needs an input `StsDestination` to get you a `StsClient`.

Reply:
renamed to `StsClientProvider`